model/annotations/task_annotations.go

```diff
@@ -210,3 +212,24 @@ func UpdateAnnotation(a *TaskAnnotation, userDisplayName string) error {
 	)
 	return errors.Wrapf(err, "problem adding task annotation for '%s'", a.TaskId)
 }
+
+// ValidateIssueURL checks the URL of an IssueLink to make sure that it has a scheme and a host.
+func ValidateIssueURL(issueURL string) error {
+	u, err := url.ParseRequestURI(issueURL)
+	if err != nil {
+		return errors.Wrapf(err, "error parsing request uri '%s'", issueURL)
+	}
+	if u.Scheme != "http" && u.Scheme != "https" {
```

Comment:
why do we need to prevent other schemes?

Reply:
The assumption was that users will be using URLs of issue tracker websites, which often use https or http. This could be changed to accept any scheme if it doesn't seem worth checking for

Reply:
We expect it to be one of the two.  If someone puts in a different scheme, I would assume it's a mistake. The error message is descriptive enough so that on the chance that a team wants to use a different scheme, they'll see what is happening and can reach out to us and it's an easy fix. Do you know of a use case where it may be something other than these two?

Reply:
I don't know of a use case. I was making the case not to restrict the scheme more than necessary, in order to avoid needing to fix it later on.
On the other hand, these are the only two schemes we'd expect and checking for one of them ensures the specified scheme is a valid one.